releases: add tidb 3.0.20 release notes (#4421)
pingcap/docs: releases/release-3.0.20.md

+ TiDB

    - Fix the resolved txn status cache for pessimistic transactions. [#21706](https://github.com/pingcap/tidb/pull/21706)
    - Fix the statistics are inaccurate when querying `INFORMATION_SCHEMA.TIDB_HOT_REGIONS` [#21319](https://github.com/pingcap/tidb/pull/21319)
    - Raise an error when preparing the `load data` statement. [#21222](https://github.com/pingcap/tidb/pull/21222)
    - `DELETE` may not delete data correctly when the database name is not in pure lower representation. [#21205](https://github.com/pingcap/tidb/pull/21205)
    - Fix stack overflow when building recursive view. [#21000](https://github.com/pingcap/tidb/pull/21000)
    - Fix an goroutine leak issue in TiKV client. [#20863](https://github.com/pingcap/tidb/pull/20863)
    - Fix wrong default zero value for year type [#20828](https://github.com/pingcap/tidb/pull/20828)
    - Avoid goroutine leak in index lookup join [#20791](https://github.com/pingcap/tidb/pull/20791)
    - Fix "insert select for update" return malformed packet in pessimistic txn [#20681](https://github.com/pingcap/tidb/pull/20681)
    - Fix unknown time zone 'posixrules'. [#20605](https://github.com/pingcap/tidb/pull/20605)

>     - Fix data too long when converting to bit returns a null value [#20362](https://github.com/pingcap/tidb/pull/20362)

Reviewer: ```suggestion
    - Fix a problem of data too long when converting from int unsigned it to bit [#20362](https://github.com/pingcap/tidb/pull/20362)
```

    - Fix corrupted default value for bit type column. [#20339](https://github.com/pingcap/tidb/pull/20339)

>     - Avoid propagate column optimization for Hybrid type. [#20296](https://github.com/pingcap/tidb/pull/20296)

Reviewer: ```suggestion
    - Fix potentially incorrect results when one of the equal condition is Enum of Set type. [#20296](https://github.com/pingcap/tidb/pull/20296)
```

    - Fix wrong behavior for != any() [#20061](https://github.com/pingcap/tidb/pull/20061)

## Others

+ TiDB

>     - Fix an issue that invalid results using BETWEEN...AND... with type conversion [#21503](https://github.com/pingcap/tidb/pull/21503)

Reviewer: This line should be moved to bug fix.

    - Deprecate `enable-streaming`. [#21054](https://github.com/pingcap/tidb/pull/21054)

>     - Fix compatibility for `ADDDATE` function [#21008](https://github.com/pingcap/tidb/pull/21008)

Reviewer: This line should be moved to bug fix.

>     - Set correct default value for new added enum column. [#20999](https://github.com/pingcap/tidb/pull/20999)

Reviewer: This line should be moved to bug fix.

    - Fix the result of SQL `select DATE_ADD('2007-03-28 22:08:28',INTERVAL "-2.-2" SECOND)` to be same with the MySQL; [#20627](https://github.com/pingcap/tidb/pull/20627)

>     - Fix #20490, incorrect default value when modify column [#20532](https://github.com/pingcap/tidb/pull/20532)

Reviewer: ```suggestion
    - Fix incorrect default value when modify column [#20532](https://github.com/pingcap/tidb/pull/20532)
```

    - Fix function timestamp() get wrong result when input argument is type of float/decimal [#20469](https://github.com/pingcap/tidb/pull/20469)

>     - Statistics: fix an unsafe lock operation and adjust some logging levels [#20424](https://github.com/pingcap/tidb/pull/20424)

Reviewer: ```suggestion
    - fix a potentially deadlock problem in statistic and adjust some logging levels [#20424](https://github.com/pingcap/tidb/pull/20424)
```

>     - Fix FLOAT data type: out of range data should not be inserted [#20251](https://github.com/pingcap/tidb/pull/20251)

Reviewer: These lines should be moved to bug fix.

> + TiKV

Reviewer 2: lgtm

>     - Add end_point_slow_log_threshold config [#9145](https://github.com/tikv/tikv/pull/9145)

Reviewer 2: Improvements

>     - Fix the issue that reports a key-exist error when a key is locked and deleted by a committed transaction. [#8931](https://github.com/tikv/tikv/pull/8931)

Reviewer 2: Bug Fixes

> + PD

Reviewer 3: LGTM

---
title: TiDB 3.0.20 Release Notes
---

# TiDB 3.0.20 Release Notes

> Release date: December 23, 2020

Reviewer 4: ```suggestion
Release date: December 25, 2020
```

TiDB version: 3.0.20

## Compatibility Change

+ TiDB

    - Deprecate the `enable-streaming` configuration item [#21054](https://github.com/pingcap/tidb/pull/21054)

## Improvements

+ TiDB

    - Raise an error when preparing the `LOAD DATA` statement [#21222](https://github.com/pingcap/tidb/pull/21222)

+ TiKV

    - Add the `end_point_slow_log_threshold` configuration item [#9145](https://github.com/tikv/tikv/pull/9145)

## Bug Fixes

+ TiDB

>     - Fix the resolved txn status cache for pessimistic transactions. [#21706](https://github.com/pingcap/tidb/pull/21706)

Reviewer 4: ```suggestion
    - Fix the incorrect cache of the transaction status for pessimistic transactions [#21706](https://github.com/pingcap/tidb/pull/21706)
```

>     - Fix the statistics are inaccurate when querying `INFORMATION_SCHEMA.TIDB_HOT_REGIONS` [#21319](https://github.com/pingcap/tidb/pull/21319)

Reviewer 4: ```suggestion
    - Fix the issue of inaccurate statistics that occurs when querying `INFORMATION_SCHEMA.TIDB_HOT_REGIONS` [#21319](https://github.com/pingcap/tidb/pull/21319)
```
